braindecode/models/usleep.py

```diff
@@ -0,0 +1,281 @@
+                          padding=padding),
+                nn.ELU(),
+                nn.BatchNorm1d(num_features=out_channels),
+            )
+
+    def forward(self, x):
+        x = self.block_prepool(x)
+        residual = x
+        if x.shape[-1] % 2:
+            x = nn.ConstantPad1d(padding=1, value=0)(x)
+        x = nn.MaxPool1d(kernel_size=self.downsample, stride=self.downsample)(x)
+        return x, residual
+
+
+class DecoderBlock(nn.Module):
+    '''Decoding block for a timeseries x of shape (B, C, T).'''
+    def __init__(self,
+                 in_channels=2,
+                 out_channels=2,
+                 kernel_size=9,
+                 upsample=2,
+                 with_skip_connection=True):
+        super().__init__()
+        self.in_channels = in_channels
+        self.out_channels = out_channels
+        self.kernel_size = kernel_size
+        self.upsample = upsample
+        self.with_skip_connection = with_skip_connection
+        padding = (kernel_size - 1) // 2   # chosen to preserve dimension
+
+        self.block_preskip = nn.Sequential(
+                    nn.Upsample(scale_factor=upsample),
+                    nn.Conv1d(in_channels=in_channels,
+                              out_channels=out_channels,
+                              kernel_size=kernel_size,
+                              padding=padding),
+                    nn.ELU(),
+                    nn.BatchNorm1d(num_features=out_channels),
+                )
+        self.block_postskip = nn.Sequential(
+                    nn.Conv1d(in_channels=(
+                            2 * out_channels if with_skip_connection else out_channels),
+                              out_channels=out_channels,
+                              kernel_size=kernel_size,
+                              padding=padding),  # to preserve dimension (check)
+                    nn.ELU(),
+                    nn.BatchNorm1d(num_features=out_channels),
+                )
+
+    def forward(self, x, residual):
+        x = self.block_preskip(x)
+        if self.with_skip_connection:
+            x, residual = crop_tensors_to_match(x, residual, axis=-1)  # in case of mismatch
+            x = torch.cat([x, residual], axis=1)  # (B, 2 * C, T)
+        x = self.block_postskip(x)
+        return x
+
+
+class USleep(nn.Module):
+    """Sleep staging architecture from [1]_.
+
+    U-Net (autoencoder with skip connections) feature-extractor for sleep staging described in [1]_.
+
+    For the encoder ('down'):
+        -- the temporal dimension shrinks (via maxpooling in the time-domain)
+        -- the spatial dimension expands (via more conv1d filters in the time-domain)
+    For the decoder ('up'):
+        -- the temporal dimension expands (via upsampling in the time-domain)
+        -- the spatial dimension shrinks (via fewer conv1d filters in the time-domain)
+    Both do so at exponential rates.
+
+    Parameters
+    ----------
+    n_channels : int
```

Comment:
we need to fix this naming

Reply:
Yes. We just changed it to in_chans for consistency with (some of the other) braindecode models. We will open a more general PR for naming.

Reply:
@l-omar-chehab in_chans now